src/EnvRegistry.ts

```diff
@@ -0,0 +1,60 @@
+import { from } from 'rxjs';
+import { switchMap } from 'rxjs/operators';
+import { messages } from '@capsulajs/capsulajs-configuration-service/lib/utils';
+import { ConfigurationService, ConfigurationServiceLocalStorage } from '@capsulajs/capsulajs-configuration-service';
+import {
+  EnvRegistry as EnvRegistryInterface,
+  EnvRegistryItem,
+  EnvironmentsRequest,
+  EnvironmentsResponse,
+  RegisterResponse,
+} from './api/EnvRegistry';
+import { envKeyValidator, envValidator, validationMessages } from './utils';
+import { ConfigEntry, EntriesResponse } from './types';
+
+export default class EnvRegistry implements EnvRegistryInterface {
+  private configurationService: ConfigurationService;
+  private repositoryCreated: boolean;
+  private readonly repository: string;
+
+  constructor(token: string) {
+    this.configurationService = new ConfigurationServiceLocalStorage(token); // add logic to choose specific provider
+    this.repositoryCreated = false;
+    this.repository = 'environmentRegistry';
+  }
+
+  public createRepository() {
```

Comment:
Why `createRepository` is public ?

Reply:
It was a problem with jest and typescript in `resgister.test`, resolved with a `@ts-ignore`